File: CodingSamples/DotNet/Runtime/DelegateTest/DemoApp/Investment.cs
delegate float Interest(int years);

class Investment
{
    public double Payment { get; }

    public int Count { get; }

    public Investment(double payment, int count)
    {
        Payment = payment;
        Count = count;
    }

    public double FutureValue(Interest rate)
    {
        float i = rate.Invoke(Count);
        return (Payment / i) * (Math.Pow(1 + i, Count) - 1);
    }
}

File: CodingSamples/Java/Runtime/FunctionalTest1/Investment.java
//functional interface
interface Interest {
	float forPeriod(int years);
}

class Investment {

	private final double payment;
	private final int count;

	public Investment(double payment, int count) {
		this.payment = payment;
		this.count = count;
	}

	public double payment() { return payment; }

	public int count() { return count; }

	public double futureValue(Interest rate) {
		float i = rate.forPeriod(count);
		return (payment / i) * (Math.pow(1 + i, count) - 1);
	}
}

